src/clusterfuzz/_internal/base/tasks.py

```diff
@@ -241,6 +242,51 @@ def get_utask_filters(is_chromium, is_linux):
   return pubsub_filter
 
 
+def get_ttask_commands_queues():
+  """Get queues tworkers should be querying for ttask commands. (i.e. tworkers
+  on Linux will need to know about the Windows and Mac queues)."""
+
+
+def get_machine_template_for_queue(queue_name):
+  """Gets the machine template for the instance used to execute a task from
+  |queue_name|. This will be used by tworkers to schedule the appropriate
+  machine using batch to execute the utask_main part of a utask."""
+  initial_queue_name = queue_name
+
+  # Handle it being high-end (preemptible) or not.
+  if queue_name.startswith(JOBS_PREFIX):
+    is_high_end = False
+    prefix = JOBS_PREFIX
+  else:
+    assert queue_name.startswith(HIGH_END_JOBS_PREFIX)
+    is_high_end = True
+    prefix = HIGH_END_JOBS_PREFIX
+  # Add 1 for hyphen.
+  queue_name = queue_name[len(prefix) + 1:]
+
+  template_name = f'clusterfuzz-{queue_name}'
+  if not is_high_end:
+    template_name = f'{template_name}-pre'
+
+  templates = get_machine_templates()
+  for template in templates:
+    if template['name'] == template_name:
+      logs.log(
+          f'Found machine template for {initial_queue_name}',
+          machine_template=template)
+      return template
+  return None
+
+
+def get_machine_templates():
+  """Returns machine templates."""
+  # TODO(metzman): Cache this.
+  clusters_config = local_config.Config(local_config.GCE_CLUSTERS_PATH).get()
```

Comment:
We may want to replace this with a more specialized format later, as I want us to get rid of the whole `manage_vms` bits.